### PR TITLE
[feat]: [CDS-83036]: adding a kubectl cli auth hint for K8s auth errors

### PR DESCRIPTION
**Pull request by markoharness**
## Describe your changes
Added additional hint for K8s auth errors by suggesting a kubectl cli auth using ${HARNESS_KUBE_CONFIG_PATH}

Screenshot 2023-11-30 at 10 03 17

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55997)

### CODE REVIEW

**Review by alebuh — DISMISSED** [review]

**Review by abhinavsingh1196 — DISMISSED** [review]

**Comment by abosii-harness** [comment]
@markoharness as part of this ticket we wanted to inform user that they can use `kubectl auth can-i` to check for permissions. We could even detect from error messages the verbs and suggest the full command

File: 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java [hunk 1]
```diff
@@ -24,8 +24,11 @@ public class KubernetesExceptionHints {
       "Please connect remotely to Harness delegate and verify network connection between Kubernetes cluster and Harness delegate.";
   public final String K8S_API_SOCKET_TIMEOUT_EXCEPTION =
       "Please connect remotely to Harness delegate and verify if Harness delegate is whitelisted to access Kubernetes API.";
+  public final String K8S_KUBECTL_AUTH_HINT =
```
[the remark this answers is not in the document]
Comment (abosii-harness) [reply 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java:27]: nvm

**Review by TarunUba1 — APPROVED** [review]

File: 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (abosii-harness) [reply 970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java:27]: but I would put this as a different hint, just for readability

**Review by abosii-harness — CHANGES_REQUESTED** [review]
Please add screenshots with testing

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by markoharness** [comment]
trigger uts

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by abosii-harness — COMMENTED** [review]
Please provide screenshots with testing

**Comment by markoharness** [comment]
*[`970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java` line 27 at r1](https://reviewable.io/reviews/harness/harness-core/55997#gh-1406249201:-NkQhxoP8VhmyV5FMYx1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/d749739c4d71c7b77a9514b47cea249befe4d6df/970-api-services-beans/src/main/java/io/harness/k8s/exception/KubernetesExceptionHints.java#L27)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

but I would put this as a different hint, just for readability
</blockquote></details>

Done.

**Comment by markoharness** [comment]
done

**Review by abosii-harness — APPROVED** [review]

**Comment by markoharness** [comment]
trigger review

**Pull request merged**